LM IR BLinker.X/main.c

```diff
@@ -94,6 +101,14 @@ void main(void)
        
     }
 }
-/**
- End of File
-*/
\ No newline at end of file
+
+
+// ========== LOCAL FUNCTIONS ==========
+void bitTimerInterrupt() {
+    PWM_EN = pattern[patternIndex];
+    
+    // Increment the pattern index and reset to 0 if you've looped back around
+    if (++patternIndex >= sizeof(pattern)) {        // ++var will increment var by one before using it

```

Comment:
An example of a "pre-increment". It works like this:

```C
a = ++b;
```

is equivalent to

```C
b = b + 1;
a = b;
```

You can do a similar thing in a "post-increment" which increments the variable **after** the operation. so `a = b++;` would be equivalent to:

```C
a = b;
b = b + 1;
```

You can also decrement in the same way with `--` instead of `++`

Reply:
This is also great, I was thinking about how I would do this in my longhand simple minded version of code which would have been using a crap ton of instruction cycles by calling out each bit: preamble1= (zero or one), preamble2, etc. then I was going to have an if function act upon that and flip bit high or low depending on a one or a zero, then have a delay after. 

Mostly just telling you this so you know I was thinking of how to do it the elementary school way, and am very happy to learn the professional way.